src/abcdmicro/build_template.py

```diff
@@ -0,0 +1,428 @@
+from __future__ import annotations
+
+import tempfile
+from collections import defaultdict
+from collections.abc import Mapping, Sequence
+from pathlib import Path
+
+import ants
+import numpy as np
+from ants.core import ANTsImage, ANTsTransform
+
+from abcdmicro.resource import InMemoryVolumeResource, VolumeResource
+from abcdmicro.util import update_volume_metadata
+
+
+def average_volumes(
+    volume_list: Sequence[VolumeResource], normalize: bool = True
+) -> VolumeResource:
```

Comment:
```suggestion
def average_volumes(
    volume_list: Sequence[VolumeResource], normalize: bool = True
) -> InMemoryVolumeResource:
```

(I guess we might as well say that it's in memory since we know for sure that it is)